integration/tests/karpenter/karpenter_test.go

```diff
@@ -0,0 +1,64 @@
+package karpenter
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	. "github.com/weaveworks/eksctl/integration/runner"
+	"github.com/weaveworks/eksctl/integration/tests"
+	clusterutils "github.com/weaveworks/eksctl/integration/utilities/cluster"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/testutils"
+)
+
+var params *tests.Params
+
+func init() {
+	testing.Init()
+	if err := api.Register(); err != nil {
+		panic(fmt.Errorf("unexpected error registering API scheme: %w", err))
+	}
+	params = tests.NewParams("")
+}
+
+func TestKarpenter(t *testing.T) {
+	testutils.RegisterAndRun(t)
+}
+
+var _ = Describe("(Integration) Karpenter", func() {
+
+	var (
+		clusterName string
+	)
+
+	BeforeEach(func() {
+		clusterName = fmt.Sprintf("it-karpenter-%d", time.Now().Unix())
+	})
+
+	AfterEach(func() {
+		cmd := params.EksctlDeleteCmd.WithArgs(
+			"cluster", clusterName,
+			"--verbose", "4",
+		)
+		Expect(cmd).To(RunSuccessfully())
+	})
+
+	Context("Creating a cluster with Karpenter", func() {
+		It("should support karpenter", func() {
+			cmd := params.EksctlCreateCmd.
+				WithArgs(
+					"cluster",
+					"--config-file=-",
+					"--verbose=4",
+				).
+				WithoutArg("--region", params.Region).
+				WithStdin(clusterutils.ReaderFromFile(clusterName, params.Region, "testdata/cluster-config.yaml"))
+
+			Expect(cmd).To(RunSuccessfully())
```

Comment:
is there any other assertion that could be added here to check that karpenter was installed successfully on the cluster?

Reply:
Hmmm, hmmm. Check pod status could be added.

Reply:
Added this one. I'm checking for Pods being healthy in `karpenter` namespace. 👍